Mathlib/NumberTheory/ModularForms/Basic.lean

```diff
@@ -339,6 +339,32 @@ def mul {k_1 k_2 : ℤ} [Γ.HasDetPlusMinusOne] (f : ModularForm Γ k_1) (g : Mo
   bdd_at_cusps' hc γ hγ := by
     simpa [mul_slash] using ((f.bdd_at_cusps' hc γ hγ).mul (g.bdd_at_cusps' hc γ hγ)).smul _
 
+
+
+/-- The modular form of weight `∑ i, k i` given by the product of `n` modular forms of
+weights `k i`. -/
+def finprod_add_weights {n : ℕ} (k : Fin n → ℤ) [Γ.HasDetPlusMinusOne]
+    (γ : (i : Fin n) → ModularForm Γ (k i)) : ModularForm Γ (∑ i, k i) := by
```

Comment:
I'd suggest perhaps using a letter other than `γ`. (The context would suggest `γ` is a generic element of `Γ` or something.) Maybe `f`? I think we should also have a simp lemma saying that the underlying function of `finprod_add_weights k γ` is the product of the underlying functions.